src/pyorcidator/quickstatements.py

```diff
@@ -0,0 +1,123 @@
+"""A data model for quickstatements."""
+
+import datetime
+from typing import List, Union
+
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated, Literal
+
+__all__ = [
+    "EntityQualifier",
+    "DateQualifier",
+    "TextQualifier",
+    "Qualifier",
+    "CreateLine",
+    "TextLine",
+    "EntityLine",
+    "Line",
+]
+
+
+class EntityQualifier(BaseModel):
+    """A qualifier that points to Wikidata entity."""
+
+    type: Literal["Entity"] = "Entity"
+    predicate: str = Field(regex=r"^[PQS]\d+$")
+    target: str = Field(regex=r"^[PQS]\d+$")
+
+    def get_target(self):
+        return self.target
+
+
+class DateQualifier(BaseModel):
+    """A qualifier that points to a date string."""
+
+    type: Literal["Date"] = "Date"
+    predicate: str = Field(regex=r"^[PQS]\d+$")
+    target: str
+
+    def get_target(self):
+        return self.target
+
+    @classmethod
+    def start_time(cls, target: Union[str, datetime.datetime]) -> "DateQualifier":
+        return cls(predicate="P580", target=cls._handle_date(target))
+
+    @classmethod
+    def end_time(cls, target: Union[str, datetime.datetime]) -> "DateQualifier":
+        return cls(predicate="P582", target=cls._handle_date(target))
+
+    @staticmethod
+    def _handle_date(target: Union[str, datetime.datetime]) -> str:
+        if isinstance(target, str):
+            return target
+        elif isinstance(target, datetime.datetime):
+            # TODO implement logic for creating correct format like
```

Comment:
I added testing but @jvfe implementing this line might be a nice small task for you. Since you've been commenting I am under the assumption you want to make contributions, right?

Reply:
Sorry, it was my intention to implement tests for this module but most of my day I spent away from the keyboard (national holiday in Brazil) and you guys are moving pretty quick. But no worries!

Reply:
Though the string formatting for the date could be something like `target.strftime("+%Y-%m-%dT00:00:00Z/") + precision`. There would just have to be some logic to say if the precision of the date is day (11), month (10) or year (9).